pkg/controller/publishingstrategy/publishingstrategy_controller.go

```diff
@@ -261,5 +266,138 @@ func (r *ReconcilePublishingStrategy) Reconcile(request reconcile.Request) (reco
+						log.Info(fmt.Sprintf("%s already exists on cluster", ingressController.Name))
+						continue
+					} else { // if the ingresscontroller name is not default and the contents do not match
+						// create a new ingresscontroller
+						publishingStrategyIngressDNSName := publishingStrategyIngress.DNSName
+						firstPeriodIndex := strings.Index(publishingStrategyIngressDNSName, ".")
+						newIngressControllerName := publishingStrategyIngressDNSName[:firstPeriodIndex]
+						newIngressController, err := newApplicationIngressControllerCR(newIngressControllerName, string(publishingStrategyIngress.Listening), publishingStrategyIngress.DNSName, newCertificate)
+						if err != nil {
+							log.Error(err, "failed to generate data for new ingresscontroller CR")
+						}
+						err = r.client.Create(context.TODO(), newIngressController)
+						if err != nil {
+							log.Error(err, "failed to create new ingresscontroller CR")
+							return reconcile.Result{}, err
+						}
+						log.Info(fmt.Sprintf("succesfully create new ingresscontroller %s", newIngressControllerName))
+					}
+				}
+			}
+		}
+	}
 	return reconcile.Result{}, nil
 }
+
+// newApplicationIngressControllerCR creates a new IngressController CR
+func newApplicationIngressControllerCR(ingressControllerCRName, scope, dnsName string, certificate *corev1.LocalObjectReference) (*operatorv1.IngressController, error) {
+	loadBalancerScope := operatorv1.LoadBalancerScope("")
+	switch scope {
+	case "internal":
+		loadBalancerScope = operatorv1.InternalLoadBalancer
+	case "external":
+		loadBalancerScope = operatorv1.ExternalLoadBalancer
+	default:
+		return &operatorv1.IngressController{}, errors.New("ErrCreatingIngressController")
+	}
+
+	return &operatorv1.IngressController{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ingressControllerCRName,
+			Namespace: ingressControllerNamespace,
+		},
+		Spec: operatorv1.IngressControllerSpec{
+			DefaultCertificate: certificate,
+			Domain:             dnsName,
+			EndpointPublishingStrategy: &operatorv1.EndpointPublishingStrategy{
+				Type: operatorv1.LoadBalancerServiceStrategyType,
+				LoadBalancer: &operatorv1.LoadBalancerStrategy{
+					Scope: loadBalancerScope,
+				},
+			},
+		},
+	}, nil
+}
+
+// doesIngressMatch checks if application ingress in PublishingStrategy CR matches with IngressController CR
+func doesIngressMatch(publishingStrategyIngress *cloudingressv1alpha1.ApplicationIngress, ingressController *operatorv1.IngressController) bool {
+	if string(publishingStrategyIngress.Listening) != string(ingressController.Spec.EndpointPublishingStrategy.Type) {
+		return false
+	}
+	if publishingStrategyIngress.DNSName != ingressController.Spec.Domain {
+		return false
+	}
+	if publishingStrategyIngress.Certificate.Name != ingressController.Spec.DefaultCertificate.Name {
+		if publishingStrategyIngress.Certificate.Namespace != ingressController.Namespace {
+			return false
+		}
```

Comment:
This logic should probably not be nested in this if loop. The fill loop will return false regardless of what happens in this if block.